src/tools/build_helper/src/ci.rs

```diff
@@ -19,6 +19,15 @@ impl CiEnv {
     pub fn is_ci() -> bool {
         Self::current() != CiEnv::None
     }
+
+    /// Checks if running in upstream-managed CI job.
+    pub fn is_upstream_ci_job() -> bool {
+        Self::is_ci()
+            // If both are present, we can assume it's an upstream CI job
+            // as they are always set unconditionally.
+            && std::env::var_os("CI_JOB_NAME").is_some()
+            && std::env::var_os("TOOLSTATE_REPO").is_some()
```

Comment:
I could use `git config --get remote.origin.url` instead of this if someone can confirm whether we run CI on any repository other than `rust-lang/rust`.